python/cuml/cuml/tests/test_spectral_embedding.py

```diff
@@ -0,0 +1,260 @@
+# Copyright (c) 2025, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import os
+from urllib.request import urlretrieve
+
+import cupy as cp
+import h5py
+import numpy as np
+import pytest
+from sklearn.datasets import fetch_openml, make_s_curve
+from sklearn.manifold import SpectralEmbedding as skSpectralEmbedding
+from sklearn.manifold import trustworthiness
+from sklearn.model_selection import train_test_split
+
+from cuml.manifold import SpectralEmbedding
+
+# Test parameters
+N_NEIGHBORS = 15
+N_COMPONENTS = 2
+
+
+def download_fashion_mnist(cache_dir="/tmp/rapids_datasets"):
```

Comment:
Do we need to use fashion mnist for the tests here? Is there another dataset we already download (or one from `sklearn.datasets`) that could work as well? Or one that we could generate?

There's an added cost to downloading data in CI (https://github.com/rapidsai/cuml/issues/6558) - these downloads can fail, and they tie our test suite to remove datasources continuing to exist. This one in particular also requires `h5py`, a library we don't explicitly depend on currently. IMO it'd be nice to find an alternative way to test.

_IF_ we do end up using this for testing, please move it to be a fixture in `cuml.tests.conftest`.